controllers/apps/transformer_component_vars.go

```diff
@@ -174,25 +175,34 @@ func createOrUpdateEnvConfigMap(ctx graph.TransformContext, dag *graph.DAG,
 		return mergeWith(nil)
 	}()
 
+	newObj := builder.NewConfigMapBuilder(envKey.Namespace, envKey.Name).
+		// Priority: dynamic < static < built-in
+		AddLabelsInMap(synthesizedComp.DynamicLabels).
```

Comment:
We only apply dynamic labels and annotations to resources that may be exposed to user access or management, such as pods, services and account secrets.